From 3280c5ec9940f329dc75b2b8d9a52285b20209f0 Mon Sep 17 00:00:00 2001
From: Johannes Zarl-Zierl <johannes@zarl-zierl.at>
Date: Wed, 5 May 2021 00:09:49 +0200
Subject: [PATCH] Use plain Marble package instead of KexiMarble

Marble has shipped with a MarbleConfig.cmake file since ~2016. Kexi
already depends on KF5 versions that are much newer than that.

* asturm 2021-09-14: Backported to 3.2.0.
  - s/KEXI/Kexi/ in set_package_properties(Marble ...) to fix patch context
  - partially merging d09be29f56b94d6a522fa30a5b661926baca973c (drop MARBLE_MIN_VERSION)

---
 CMakeLists.txt                                |   8 +-
 src/plugins/forms/widgets/CMakeLists.txt      |   2 +-
 .../forms/widgets/mapbrowser/CMakeLists.txt   |   3 +-
 3 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 97c58fa10..548b2708f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -196,17 +196,14 @@ endif()
 ##
 ## Test for marble
 ##
-set(MARBLE_MIN_VERSION "0.19.2")
-find_package(KexiMarble ${MARBLE_MIN_VERSION})
-set_package_properties(KexiMarble PROPERTIES
+find_package(Marble CONFIG)
+set_package_properties(Marble PROPERTIES
     DESCRIPTION "KDE World Globe Widget library"
     URL "https://marble.kde.org"
     TYPE RECOMMENDED
     PURPOSE "Required by Kexi form map widget"
 )
-if(NOT MARBLE_FOUND)
-    set(MARBLE_INCLUDE_DIR "")
-else()
+if(Marble_FOUND)
     set(HAVE_MARBLE TRUE)
 endif()
 set_package_properties(GLIB2 PROPERTIES TYPE RECOMMENDED PURPOSE "${_REQUIRED_BY_MDB}")
diff --git a/src/plugins/forms/widgets/CMakeLists.txt b/src/plugins/forms/widgets/CMakeLists.txt
index 107d578a8..109341fe0 100644
--- a/src/plugins/forms/widgets/CMakeLists.txt
+++ b/src/plugins/forms/widgets/CMakeLists.txt
@@ -11,7 +11,7 @@ endmacro()
 # the main widgets plugin
 add_subdirectory(main)
 
-if(MARBLE_FOUND)
+if(Marble_FOUND)
 #TODO  add_subdirectory(mapbrowser)
 endif()
 
diff --git a/src/plugins/forms/widgets/mapbrowser/CMakeLists.txt b/src/plugins/forms/widgets/mapbrowser/CMakeLists.txt
--- a/src/plugins/forms/widgets/mapbrowser/CMakeLists.txt
+++ b/src/plugins/forms/widgets/mapbrowser/CMakeLists.txt
@@ -1,7 +1,6 @@
 include_directories(
     ${CMAKE_SOURCE_DIR}/src/formeditor
     ${CMAKE_SOURCE_DIR}/src/core
-    ${MARBLE_INCLUDE_DIR}
 )
 
 set(kexiforms_mapwidgetplugin_SRCS
@@ -20,8 +19,8 @@ target_link_libraries(kexiforms_mapwidgetplugin
     kexiextendedwidgets
     kexiformutils
     kexidataviewcommon
-    ${MARBLE_LIBRARIES}
 
+    Marble
     Qt5::Xml
 )
 
-- 
GitLab

